sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/javaCode.scala

```diff
@@ -118,10 +119,10 @@ object JavaCode {
 /**
  * A trait representing a block of java code.
  */
-trait Block extends JavaCode {
+trait Block extends TreeNode[Block] with JavaCode {
   import Block._
 
-  // The expressions to be evaluated inside this block.
+  // All expressions to be evaluated inside this block and underlying blocks.
   def exprValues: Set[ExprValue]
```

Comment:
do we still need it?

Reply:
We can remove it, as discussed in other PR previously.